### PR TITLE
Merge New Track Idea Into Coupler Track.

### PR DESCRIPTION
**Pull request by Cream-Tea**
I accidentally deleted the previous pull request, was figuring out what the eclipse files are, I hope they don't affect it. The changes are basically to Locomotive behaviour (They couple when they are set to), TrackCoupler, the modifed texture and the EnumTrack to make way for 6 icons instead of 4.

### CODE REVIEW

**Comment by AEnterprise** [comment]
@Cream-Tea you don't have to do a new pr each time you make changes, if you push changes the pr will update, also you might wana sqach your commits and then force a push (again pr will update automaticly)

**Comment by Cream-Tea** [comment]
Alright thanks, I am concerned about all those eclipse files, and I don't seem to be able to get rid of them.

**Comment by Vexatos** [comment]
The problem is that you did everything in one commit and you added the gitignore file after the commit. Of course it won't be ignored if the files have already been added.

**Comment by Cream-Tea** [comment]
Alright I think I've got rid of them now.

**Comment by Cream-Tea** [comment]
I think that's it.

**Comment by Vexatos** [comment]
Hmmm... `readyToLink` is not being stored through server reboots?

**Comment by Cream-Tea** [comment]
Oh yeah, because we'd need to make sure it kept its state, how would I do that? Through NBT tags?

**Comment by Vexatos** [comment]
writeToNBT and readFromNBT

**Comment by Cream-Tea** [comment]
Alright I'll attempt that.

**Comment by CovertJaguar** [comment]
This looks good enough for now. I might tweak the code a bit, but I'd say expect it in the next release.

**Comment by CovertJaguar** [comment]
Correction...you seem to have deleted/broken the lang submodule. Please remove that from the PR.

File: lang [hunk 1]
```diff
@@ -1 +0,0 @@
-Subproject commit 1cbf24e9375b95f641cc4b302be7d60c54317217
```
Comment (CovertJaguar) [inline lang]: Please unstage/revert this...I'm not sure how to do it from my end but it breaks things.

**Comment by CovertJaguar** [comment]
Eh, nvm I think figured out how to fix the merge.

**Comment by CovertJaguar** [comment]
Ugh...ok I'm going to have to just pick out the file changes directly I think. Your history is completely buggered.

**Comment by CovertJaguar** [comment]
Ok, I got it figured out, but I'll have to finish my tweaks later. Don't worry about messing with this.

**Comment by Cream-Tea** [comment]
Alright thanks, sorry about that, but once you've finished tweaking it I'll be able to download it and "refresh" my fork.

**Comment by CovertJaguar** [comment]
See #741

**Comment by CovertJaguar** [comment]
Alright, it should all be working. Thanks for your efforts!

Let me know if you see any issues.

**Pull request closed without merging**